compiler/rustc_lint/src/drop_forget_useless.rs

```diff
@@ -163,10 +163,24 @@ impl<'tcx> LateLintPass<'tcx> for DropForgetUseless {
                     );
                 }
                 sym::mem_drop if is_copy && !drop_is_single_call_in_arm => {
+                    let sugg = if let Some((_, node)) = cx.tcx.hir().parent_iter(expr.hir_id).nth(0)
+                        && let Node::Stmt(stmt) = node
+                        && let StmtKind::Semi(e) = stmt.kind
+                        && e.hir_id == expr.hir_id
+                        && let Ok(value) = cx.sess().source_map().span_to_snippet(arg.span)
+                    {
+                        DropCopySuggestion::Suggestion {
+                            sugg: expr.span,
+                            replace: format!("let _ = {}", value),
+                        }
```

Comment:
> I don't think we need to use `span_to_snippet` here, we can take the [Span](https://doc.rust-lang.org/beta/nightly-rustc/rustc_span/struct.Span.html#method.shrink_to_lo) of the expression and the arg and do some span arithmetic to avoid it. My idea is to get the span of `drop(` and the span of `)` and in the same suggestion replace the first part by `let _ = ` and the second by nothing.
> 
> That gives better output, since the diagnostic machinery can print more targeted diff.

Your suggestions are  much better, thank you very much.